pyphare/pyphare/pharein/mhd_model.py

```
@@ -30,14 +77,141 @@ def __init__(
+                "MHDModel: a0 (B0 from vector potential) is exclusive with b0x/b0y/b0z"
+            )
+        if b0_time_dependent and not b0_from_potential:
+            # component-mode time-dependent B0: b0* are f(x,t); build space-only views at the
+            # initial time (t=0) so the total field B = B0(.,0) + B1 can still be assembled below.
+            b0x = self.st_defaulter(b0x, 0.0)
+            b0y = self.st_defaulter(b0y, 0.0)
+            b0z = self.st_defaulter(b0z, 0.0)
+            db0x_dt = self.st_defaulter(db0x_dt, 0.0)
+            db0y_dt = self.st_defaulter(db0y_dt, 0.0)
+            db0z_dt = self.st_defaulter(db0z_dt, 0.0)
+            b0x_s = lambda *xyz: b0x(*xyz, 0.0)
```

Ruff (0.15.20)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `b0x_s` as a `def`

(E731)